content/algorithms/planarity/planarity.md

```diff
@@ -0,0 +1,939 @@
+# CHECK PLANARITY
+
+## Planarity Algorithm in NetworkX
+
+## Introduction
+A graph is said to be **planar** if it can be drawn on a plane without any of its edges crossing. The `check_planarity` function in NetworkX helps determine whether a given graph is planar and provides either an embedding or a counterexample.
+
+### Mathematical Background
+A graph is **planar** if and only if it does not contain a subgraph homeomorphic to **K₅ (complete graph on 5 vertices)** or **K₃,₃ (complete bipartite graph with partition sizes 3 and 3)**. This is known as **Kuratowski’s theorem**.
+
+The planarity check in NetworkX is based on the **Left-Right Planarity Test**, an efficient combinatorial method to determine planarity.
+
+---
+
+
+
+***What is  Left-Right Planarity Test***
+
+The Left-Right Planarity Test is a linear-time algorithm used to check whether a graph is planar—that is, drawable in the plane without edge crossings. This test forms the core of the networkx.check_planarity() function.
+
+***How It Works (Conceptually):***
+
+
+
+1. *Depth-First Search (DFS):*  
+The algorithm starts by building a DFS tree of the input graph, which helps classify edges into tree edges and back edges.
+2. *Lowpoints and Embedding Constraints:*  
+For each vertex, the algorithm computes "lowpoints"—the smallest reachable ancestor in DFS. It uses this to decide how back edges can be placed to the left or right of tree edges without crossings.
+3. *Conflict Detection:*  
+The algorithm tracks embedding constraints between edges. If it detects a conflict (e.g., an edge that can't be placed without violating planarity), it concludes the graph is non-planar.
+4. *Constructs a Planar Embedding or Finds Obstruction:*  
+If the graph is planar, it returns a combinatorial embedding (rotation system). If not, it can identify a Kuratowski subgraph (e.g., K₅ or K₃,₃).
+
+
+
+
+
+
+```python
+from functools import singledispatch
```

Comment:
This line of code is not needed/used in the notebook. Try to remove any spurious cells and code. :)